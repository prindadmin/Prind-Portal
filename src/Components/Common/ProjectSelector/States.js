export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const LOADING_ERROR = 'LOADING_ERROR'
export const UPDATE_ERROR = 'UPDATE_ERROR'
export const PROJECT_TYPE_SELECTOR_OPEN = 'PROJECT_TYPE_SELECTOR_OPEN'
