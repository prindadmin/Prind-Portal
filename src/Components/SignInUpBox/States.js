export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS'
export const PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED'
