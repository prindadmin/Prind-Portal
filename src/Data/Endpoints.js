export const DEFAULTPAGE = ""
export const DEFAULTLOGGEDINPAGE = "/inception"
export const PROJECTDETAILSPAGE = "/project"
export const SIGNINPAGE = "/signin"
export const SIGNUPPAGE = "/signup"
export const MEMBERPAGE = "/team"
