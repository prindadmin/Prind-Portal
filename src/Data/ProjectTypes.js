import * as Strings from './Strings'

const ProjectTypesList = [
  {
    id: 'CDM2015Project',
    icon: 'building',
    name: Strings.PROJECT_TYPE_CDM2015PROJECT_NAME,
    description: Strings.PROJECT_TYPE_CDM2015PROJECT_DESCRIPTION
  },
  {
    id: 'DHSFProject',
    icon: 'building',
    name: Strings.PROJECT_TYPE_DHSFPROJECT_NAME,
    description: Strings.PROJECT_TYPE_DHSFPROJECT_DESCRIPTION
  }
]

export default ProjectTypesList
