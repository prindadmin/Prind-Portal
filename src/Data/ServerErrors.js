
export const PROJECT_NOT_FOUND = "PROJECT_NOT_FOUND"
export const INSUFFICIENT_PERMISSION = "INSUFFICIENT_PERMISSION"
