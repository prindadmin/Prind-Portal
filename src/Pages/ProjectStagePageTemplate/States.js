export const QUIESCENT = 'QUIESCENT'

export const CONTENT_FETCH_IN_PROGRESS = 'CONTENT_FETCH_IN_PROGRESS'
export const CONTENT_FETCH_SUCCESSFUL = 'CONTENT_FETCH_SUCCESSFUL'
export const CONTENT_FETCH_FAILED = 'CONTENT_FETCH_FAILED'
