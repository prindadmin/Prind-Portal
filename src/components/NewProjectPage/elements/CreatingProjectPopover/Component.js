import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
