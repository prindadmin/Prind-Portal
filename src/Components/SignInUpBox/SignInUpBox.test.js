import React from 'react';
import { shallow } from 'enzyme';
import ReactGA from 'react-ga';
import Component from './SignInUpBox'
import * as States from '../../States'
import * as Endpoints from '../../Data/Endpoints'

beforeAll(() => {
  ReactGA.initialize('dummy', { testMode: true });
});

it('Should render sign in', () => {
  const mockHistoryPush = jest.fn()
  const props = {
    isSignIn: true,
    isSignUp: false,
    isForgotPassword: false,
    isPasswordReset: false,
    username: "testing@buildingim.com",
    history: {
      push: mockHistoryPush
    },
    location: {
      pathname: '/signin'
    },
    auth: {
      isSignedIn: States.AUTH_UNKNOWN
    },
    user: {
      currentRoute: '/profile'
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Should render sign up', () => {
  const mockHistoryPush = jest.fn()
  const props = {
    isSignIn: false,
    isSignUp: true,
    isForgotPassword: false,
    isPasswordReset: false,
    username: "testing@buildingim.com",
    history: {
      push: mockHistoryPush
    },
    location: {
      pathname: '/signin'
    },
    auth: {
      isSignedIn: States.AUTH_UNKNOWN
    },
    user: {
      currentRoute: '/profile'
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Should render forgot password', () => {
  const mockHistoryPush = jest.fn()
  const props = {
    isSignIn: false,
    isSignUp: false,
    isForgotPassword: true,
    isPasswordReset: false,
    username: "testing@buildingim.com",
    history: {
      push: mockHistoryPush
    },
    location: {
      pathname: '/signin'
    },
    auth: {
      isSignedIn: States.AUTH_UNKNOWN
    },
    user: {
      currentRoute: '/profile'
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Should render password reset', () => {
  const mockHistoryPush = jest.fn()
  const props = {
    isSignIn: false,
    isSignUp: false,
    isForgotPassword: false,
    isPasswordReset: true,
    username: "testing@buildingim.com",
    history: {
      push: mockHistoryPush
    },
    location: {
      pathname: '/signin'
    },
    auth: {
      isSignedIn: States.AUTH_UNKNOWN
    },
    user: {
      currentRoute: '/profile'
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Should render sign in, change props to logged in, and update', () => {
  const mockHistoryPush = jest.fn()
  var props = {
    isSignIn: true,
    isSignUp: false,
    isForgotPassword: false,
    isPasswordReset: false,
    username: "testing@buildingim.com",
    history: {
      push: mockHistoryPush
    },
    location: {
      pathname: '/signin'
    },
    auth: {
      isSignedIn: States.AUTH_SUCCESS
    },
    user: {
      currentRoute: '/profile'
    }
  };
  const prevProps = props
  const component = shallow(<Component {...props} />);
  props.location.pathname = '/profile'
  component.setProps(props)
  component.instance().componentDidUpdate(prevProps)
  expect(component).toMatchSnapshot();
});


it('Should render sign in then cycle through all form options', () => {
  const mockHistoryPush = jest.fn()
  const props = {
    isSignIn: true,
    isSignUp: false,
    isForgotPassword: false,
    isPasswordReset: false,
    username: "testing@buildingim.com",
    history: {
      push: mockHistoryPush
    },
    location: {
      pathname: '/signin'
    },
    auth: {
      isSignedIn: States.AUTH_UNKNOWN
    },
    user: {
      currentRoute: '/profile'
    }
  };
  const component = shallow(<Component {...props} />);
  component.instance().toggleVisibleForm(States.SIGNINFORM)
  expect(mockHistoryPush).toHaveBeenCalledWith(Endpoints.SIGNINPAGE)
  component.instance().toggleVisibleForm(States.SIGNUPFORM)
  expect(mockHistoryPush).toHaveBeenCalledWith(Endpoints.SIGNUPPAGE)
  component.instance().toggleVisibleForm(States.FORGOTPASSWORDFORM)
  expect(mockHistoryPush).toHaveBeenCalledWith(Endpoints.FORGOTPASSWORDPAGE)
});


it('Should render sign in then show default box because provided form does not exist', () => {
  const mockHistoryPush = jest.fn()
  const props = {
    isSignIn: true,
    isSignUp: false,
    isForgotPassword: false,
    isPasswordReset: false,
    username: "testing@buildingim.com",
    history: {
      push: mockHistoryPush
    },
    location: {
      pathname: '/signin'
    },
    auth: {
      isSignedIn: States.AUTH_UNKNOWN
    },
    user: {
      currentRoute: '/profile'
    }
  };
  const component = shallow(<Component {...props} />);
  component.instance().toggleVisibleForm("NON-EXISTENT FORM")
  expect(mockHistoryPush).toHaveBeenCalledWith(Endpoints.SIGNINPAGE)
});
