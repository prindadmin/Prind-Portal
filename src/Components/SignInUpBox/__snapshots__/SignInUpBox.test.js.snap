// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render forgot password 1`] = `
<div
  id="component-sign-in-up-box"
>
  <Suspense
    fallback={<Spinner />}
  >
    <div
      className="component-sign-in-up-box-content-container"
    >
      <withRouter(Connect(ForgotPasswordBox))
        toggleVisibleForm={[Function]}
      />
    </div>
  </Suspense>
</div>
`;

exports[`Should render password reset 1`] = `
<div
  id="component-sign-in-up-box"
>
  <Suspense
    fallback={<Spinner />}
  >
    <div
      className="component-sign-in-up-box-content-container"
    >
      <withRouter(Connect(ResetPassword))
        toggleVisibleForm={[Function]}
      />
    </div>
  </Suspense>
</div>
`;

exports[`Should render sign in 1`] = `
<div
  id="component-sign-in-up-box"
>
  <Suspense
    fallback={<Spinner />}
  >
    <div
      className="component-sign-in-up-box-content-container"
    >
      <withRouter(Connect(SignInBox))
        toggleVisibleForm={[Function]}
        username="testing@buildingim.com"
      />
    </div>
  </Suspense>
</div>
`;

exports[`Should render sign in, change props to logged in, and update 1`] = `
<div
  id="component-sign-in-up-box"
>
  <Suspense
    fallback={<Spinner />}
  >
    <div
      className="component-sign-in-up-box-content-container"
    >
      <withRouter(Connect(SignInBox))
        toggleVisibleForm={[Function]}
        username="testing@buildingim.com"
      />
    </div>
  </Suspense>
</div>
`;

exports[`Should render sign up 1`] = `
<div
  id="component-sign-in-up-box"
>
  <Suspense
    fallback={<Spinner />}
  >
    <div
      className="component-sign-in-up-box-content-container"
    >
      <withRouter(Connect(SignUpBox))
        toggleVisibleForm={[Function]}
      />
    </div>
  </Suspense>
</div>
`;
